diff.py

```diff
@@ -3079,29 +3076,66 @@ def align_diffs(
         diff_lines = [
             (base, new, old if old != new else empty) for base, new, old in diff_lines
         ]
+        headers = (
+            Text("TARGET"),
+            Text(f"{padding}CURRENT ({new_diff.score})"),
+            Text(f"{padding}PREVIOUS ({old_diff.score})"),
+        )
+        current_score = new_diff.score
+        max_score = new_diff.max_score
+        previous_score = old_diff.score
     elif config.diff_mode in (DiffMode.SINGLE, DiffMode.SINGLE_BASE):
         header = Text("BASE" if config.diff_mode == DiffMode.SINGLE_BASE else "CURRENT")
-        meta = TableMetadata(
-            headers=(header,),
-            current_score=0,
-            max_score=0,
-            previous_score=None,
-        )
         diff_lines = [(line,) for line in new_diff.lines]
+        headers = (header,)
+        # Scoring is disabled for view mode
+        current_score = 0
+        max_score = 0
+        previous_score = None
     else:
-        meta = TableMetadata(
-            headers=(
-                Text("TARGET"),
-                Text(f"{padding}CURRENT ({new_diff.score})"),
-            ),
-            current_score=new_diff.score,
-            max_score=new_diff.max_score,
-            previous_score=None,
-        )
         diff_lines = [(line, line) for line in new_diff.lines]
+        headers = (
+            Text("TARGET"),
+            Text(f"{padding}CURRENT ({new_diff.score})"),
+        )
+        current_score = new_diff.score
+        max_score = new_diff.max_score
+        previous_score = None
     if config.compress:
         diff_lines = compress_matching(diff_lines, config.compress.context)
-    return meta, diff_lines
+
+    def diff_lines_to_table_lines(
+        diff_lines: List[Tuple[OutputLine, ...]]
+    ) -> List[TableLine]:
+        table_lines: List[TableLine] = []
+        for line in diff_lines:
+
+            cells: List[Tuple[Text, Optional["Line"]]] = []
+            for i in range(0, len(line)):
+                if i == 0:
+                    assert line[i].base is not None
+                    base: Text = line[i].base  # type: ignore
+                    cells.append((base, line[i].line1))
+                else:
+                    cells.append((line[i].fmt2, line[i].line2))
+
+            assert line[0].key2 is not None
+            table_lines.append(
+                TableLine(
+                    key=line[0].key2,
+                    is_data_ref=line[0].is_data_ref,
```

Comment:
(we were using [1] before, and now [0]. I think they're probably equally correct but I'll think a little more about this)